components/motor/dimensionengineering/sabertooth.go

```diff
@@ -0,0 +1,539 @@
+// Package dimensionengineering contains implementations of the dimensionengineering motor controls
+package dimensionengineering
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/jacobsa/go-serial/serial"
+	"github.com/mitchellh/mapstructure"
+	utils "go.viam.com/utils"
+
+	"go.viam.com/rdk/components/motor"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/operation"
+	"go.viam.com/rdk/registry"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+const (
+	// https://www.dimensionengineering.com/datasheets/Sabertooth2x60.pdf
+	modelName = "de-sabertooth"
+)
+
+// controllers is global to all instances, mapped by serial device.
+var (
+	globalMu    sync.Mutex
+	controllers map[string]*controller
```

Comment:
why would you need to store all controller globally? can multiple motor share the same controller? The motor close function would delete the controller from the map

Reply:
Yes, a single controller has 2 channels, so it can support up to 2 motors.